base/bool.jl

```diff
@@ -70,6 +70,74 @@ julia> [true; true; false] .⊻ [true; false; false]
 """
 xor(x::Bool, y::Bool) = (x != y)
 
+"""
+    nand(x, y)
+    ⊼(x, y)
+
+Bitwise nand (not and) of `x` and `y`. Implements
+[three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic),
+returning [`missing`](@ref) if one of the arguments is `missing`.
+
+The infix operation `a ⊼ b` is a synonym for `nand(a,b)`, and
+`⊼` can be typed by tab-completing `\\nand` or `\\barwedge` in the Julia REPL.
+
+# Examples
+```jldoctest
+julia> nand(true, false)
+true
+
+julia> nand(true, true)
+false
+
+julia> nand(true, missing)
+missing
+
+julia> false ⊼ false
+true
+
+julia> [true; true; false] .⊼ [true; false; false]
+3-element BitVector:
+ 0
+ 1
+ 1
+```
+"""
+nand(x::Bool, y::Bool) = ~(x & y)
```

Comment:
Have you tried this with just defining a single method as proposed in #40272? I really don't think you need to define 16 methods here.

Reply:
I tried it with just the single method but ran into issues when testing `nand(missing)` and `nand(x)` where `x` is an integer. Instead of defining 16 methods, I've defined 4:
```julia
julia> methods(nand)

# 4 methods for generic function "nand":
[1] nand(::Missing) in Base at missing.jl:101
[2] nand(a::BigInt, b::BigInt) in Base.GMP at gmp.jl:551
[3] nand(x::Integer) in Base at operators.jl:579
[4] nand(x, y) in Base at bool.jl:105
```
This passes all the tests I've made.

Reply:
I would skip the one-arg ones as well and just have the two-arg version. `nand(x) = x` seems questionable to me anyways.

Reply:
I understand how `nand(x) = x` seems questionable, but out of curiosity, what's your thinking on `nand(missing)`? In my limited knowledge, it seems like we should include it since the majority of other operators have the same property. If not, is there a concept I'm missing / misunderstanding?

Reply:
For any arbitrary set of values, we can decompose these two functions essentially just directly into their constituent parts:
```julia
nand(x...) = ~(&)(x...)
nor(x...) = ~|(x...)
```

Reply:
My initial thought was to edge on the more conservative side here and just implement the most minimal version for now. This implementation now seems like a sensible extension to me though, so we might as well go with that.

Reply:
Got it, appreciate your help!